README.md

```diff
@@ -99,20 +86,56 @@ Note that you may not get cached results if you use a different `nix` version th
 
 ## Building arbitrary packages
 
-The [`survey`](./survey) directory maintains a select set of Haskell executables that are known and not known to work with this approach; contributions are welcome to grow the set of working executables.
-Run for example:
+The [`survey/default.nix`](./survey/default.nix) was originally a survey of Haskell executables that are known to (and known _not_ to) work with this approach, however it also exposes packages sets with overridden Haskell packages and dependencies that you can use to build _your own_ packages. The name `survey` shouldn't put you off from using it.
+
+If you are a nix user, you can `import` this functionality and override the `haskellPackages` to include your own package, for example in the [PostgREST project](https://github.com/PostgREST/postgrest/blob/main/nix/static-haskell-package.nix).
+
+### Structure of `survey/default.nix`
+
+The process of building up to the final `haskellPackages` is broken down into multiple steps, with the intermediate package sets also exposed for your use.
+
+We start with a plain nixpkgs named `normalPkgs`, defaulting to the version provided by [`nixpkgs.nix`](./nixpkgs.nix), but you can pass in your own version here.
+You could instead provide `overlays`, which get applied to `normalPkgs`.
+The `.pkgsMusl` from `normalPkgs` now forms our base `pkgs`.
+
+Packages in `pkgsMusl` typically [only include `.so` files](https://github.com/NixOS/nixpkgs/issues/61575), but not `.a` files. We create a `archiveFilesOverlay`, which overrides our Haskell dependencies (i.e. C libraries) to include `.a` files as well. 
+This usually involves low-level library specific actions, but the goal is to upstream these to nixpkgs under a `dontDisableStatic` attribute.
+We apply this overlay to `pkgs` to get `pkgsWithArchiveFiles`.
+
+This package set _may_ be sufficient for your needs if you have a separate build system (e.g. Bazel) and are only after a statically linked GHC and library dependencies.
```

Comment:
I'm not sure if this is actually true for GHC >= 9.6 as we now have the `fixGhc` step that only gets applied in `pkgsWithStaticHaskellBinaries` (to deal with DWARF and Hadrian).

The Skyscope example that uses `pkgsWithArchiveFiles` I believe is only using GHC 9.0.2, so isn't affected.
https://github.com/tweag/skyscope/pull/94/files#diff-8cce62c1be86c722f621f909a18b4a5df8eef70b6399b40194894ad9ce09f0abR20

Do you remember if there was a reason why the `fixGhc` was not applied earlier? (I intend to try moving it earlier and seeing if the derivation changes/anything breaks).

Reply:
> Do you remember if there was a reason why the `fixGhc` was not applied earlier? (I intend to try moving it earlier and seeing if the derivation changes/anything breaks).

It look like moving it earlier should work fine and makes sense.

Reply:
Great, I intend to leave this white lie in the documentation and we can have a separate PR to move the `fixGhc` step earlier up.

Reply:
I've just noticed that at the moment `pkgsWithArchiveFiles` is slightly more broken than I realised.
I think the `compiler` override only gets applied in the `haskellLibsReadyForStaticLinkingOverlay`.

```nix
nix-repl> pkgs = import ./survey/default.nix { compiler = "ghc962"; }

nix-repl> pkgs.pkgsWithArchiveFiles.ghc
«derivation /nix/store/pd3h7k2f9p9290dhfch2wnwkp9714xgk-ghc-musl-9.2.7.drv»

nix-repl> pkgs.haskellPackagesWithLibsReadyForStaticLinking.ghc
«derivation /nix/store/11rnlr5qbk1pxwk4zjyb3xyz8wz2fyx6-ghc-musl-9.6.2.drv»
```

Reply:
I've had a go at moving GHC earlier here https://github.com/nh2/static-haskell-nix/pull/118.